torch/csrc/distributed/rpc/torchscript_functions.cpp

```diff
@@ -19,6 +20,17 @@ c10::intrusive_ptr<c10::ivalue::Future> rpcTorchscript(
     std::vector<c10::IValue>& stack,
     const float rpcTimeoutSeconds,
     const bool isAsyncExecution) {
+  if (torch::autograd::profiler::profilerEnabled() &&
+      !torch::distributed::rpc::RemoteProfilerManager::getInstance()
+           .isCurrentKeySet()) {
+    auto rpcAsyncJitKey = fmt::format(
+        "rpc_async_jit#({})->({})",
+        RpcAgent::getCurrentRpcAgent()->getWorkerInfo().name_,
+        dstWorkerName);
+    auto& remoteProfilerManager =
+        torch::distributed::rpc::RemoteProfilerManager::getInstance();
+    remoteProfilerManager.setCurrentKey(rpcAsyncJitKey);
```

Comment:
Do we have unit tests covering this? If not, lets add them.

Reply:
The test `call_rpc_with_profiling` in `jit/rpc_spawn` would fail without this.